lib/src/rules/use_named_const.dart

```diff
@@ -61,17 +62,13 @@ class _Visitor extends SimpleAstVisitor<void> {
       if (element is ClassElement) {
         for (final field
             in element.fields.where((e) => e.isStatic && e.isConst)) {
-          if (nodeField is FieldElement &&
-              (nodeField.enclosingElement as ClassElement).name ==
-                  element.name &&
-              nodeField.name == field.name) {
-            continue;
-          }
-          if (field.computeConstantValue() == value) {
+          if (field != nodeField &&
+              field.computeConstantValue() == value &&
+              (field.isPublic || field.library == library)) {
             rule.reportLint(node,
                 arguments: ['${element.name}.${field.name}'],
-                errorCode:
-                    const LintCode(lintName, 'Use predefined const {0}.'));
+                errorCode: const LintCode(
+                    lintName, "Try using the predefined constant '{0}'."));
```

Comment:
Sorry, I wasn't very clear before. Every error code has a required problem `message` and can optionally have a `correction` message. The problem message should state what's wrong with the code, such as "Constants should be referenced instead of duplicating their value." or "Don't inline constant fields." (or hopefully something better that you come up with). The correction message should suggest a possible way to fix the code, such as "Try using the predefined constant '{0}'.". It might look something like:
```
const LintCode(lintName,
    "Constants should be referenced instead of duplicating their value.",
    correction: "Try using the predefined constant '{0}'.")
```

Reply:
Unfortunately this new format doesn't display the expected constant in the output of dartanalyze.

Reply:
Yes, `dartanalyze` doesn't display correction messages by default. I can't remember whether `dart analyze` does. I suppose we could duplicate the information, using a message something like

"The constant '{0}' should be referenced instead of duplicating its value."